airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSchemas.java

```diff
@@ -59,4 +92,137 @@ public static void mutateTypeToArrayStandard(final JsonNode jsonNode) {
+   * jsonNode, final BiConsumer<JsonNode, List<String>> consumer) }. Takes path as argument so that
+   * the path can be passsed to the consumer.
+   *
+   * @param jsonSchemaNode - jsonschema object to traverse.
+   * @param path - path from the first call of traverseJsonSchema to the current node.
+   * @param consumer - consumer to be called at each node. it accepts the current node and the path to the node from the
+   *        root of the object passed at the root level invocation
+   */
+  // todo (cgardens) - replace with easier to understand traversal logic from SecretsHelper.
+  private static void traverseJsonSchemaInternal(final JsonNode jsonSchemaNode,
+                                         final String path,
+                                         final BiConsumer<JsonNode, String> consumer) {
+    if (!jsonSchemaNode.isObject()) {
+      throw new IllegalArgumentException(String.format("json schema nodes should always be object nodes. path: %s actual: %s", path, jsonSchemaNode));
+    }
+
+    consumer.accept(jsonSchemaNode, path);
+    // if type is missing assume object. not official JsonSchema, but it seems to be a common
+    // compromise.
+    final List<String> nodeTypes = getTypeOrObject(jsonSchemaNode);
+
+    for (final String nodeType : nodeTypes) {
+      switch (nodeType) {
+        case ARRAY_TYPE -> {
+          // todo (cgardens) - remove magic string.
+          final String newPath = path + "[*]";
+          // hit every node.
+          traverseJsonSchemaInternal(jsonSchemaNode.get(JSON_SCHEMA_ITEMS_KEY), newPath, consumer);
+//          jsonNode.forEach(node -> traverseJsonSchema(jsonSchemaNode.get(JSON_SCHEMA_ITEMS_KEY), node, newPath, consumer));
+        }
+        case OBJECT_TYPE -> {
+          final Optional<String> comboKeyWordOptional = getKeywordIfComposite(jsonSchemaNode);
+          if (jsonSchemaNode.has(JSON_SCHEMA_PROPERTIES_KEY)) {
+            for (final Iterator<Entry<String, JsonNode>> it = jsonSchemaNode.get(JSON_SCHEMA_PROPERTIES_KEY).fields(); it.hasNext();) {
+              final Entry<String, JsonNode> child = it.next();
+              // todo (cgardens) - remove magic string.
+              final String newPath = path + "." + child.getKey();
+              traverseJsonSchemaInternal(child.getValue(), newPath, consumer);
+            }
+          } else if (comboKeyWordOptional.isPresent()) {
+            for (final JsonNode arrayItem : jsonSchemaNode.get(comboKeyWordOptional.get())) {
+              traverseJsonSchemaInternal(arrayItem, path, consumer);
+            }
+          } else {
+            throw new IllegalArgumentException(
+                "malformed JsonSchema object type, must have one of the following fields: properties, oneOf, allOf, anyOf");
+          }
+        }
+        // if a value type do nothing
+        // case BOOLEAN_TYPE, NUMBER_TYPE, STRING_TYPE, NULL_TYPE -> consumer.accept(jsonNode, path);
```

Comment:
WE should consume the node here to be able to get all the secret path for example.

Reply:
this is just that the traversal logic is confusing. consumer.accept is called for every node. it is done on line 155. i double checked that the tests were passing for JsonSecretsProcessor when using this approach.

Reply:
check out `JsonScretsProcessorTest`. The test is running the old and new implementations and making sure they are the sam.e

(i will definitely clean up this logic so it is easier to understand)

Reply:
after reviewing further i think this traversal approach is okay. the other traversal approach we use in SecretsHandler is better if we are _replacing_ nodes, but since we aren't doing that--handled by JsonPath, I think the current approach is easiest to understand. I have updated the comments to make it clearer.